### PR TITLE
Feat/inference mode cli

### PR DESCRIPTION
**Pull request by amitpande74**
Changes:
• Expanded main() docstring to list every CLI flag (matches argparse exactly).
• Switched to argparse.parse_known_args() and warn on unknown args so Jupyter/Colab flags don’t crash the CLI.
• Conditional requirements are enforced only in training mode (--data_path, --model_class, --data_types). 
• Inference early-exit unchanged: runs when --pretrained_model + --artifacts + --data_path_test are all provided.
• Kept baseline evaluation blocks (evaluate_baseline_performance + RandomSurvivalForest) intact.

### CODE REVIEW

File: flexynesis/__main__.py [hunk 1]
```diff
@@ -499,83 +513,59 @@ def main():
...
+            output_layers_train[layer].to_csv(
+                os.path.join(args.outdir, '.'.join([args.prefix, 'train_decoded', layer, 'csv'])),
+                header=True
+            )
         for layer in output_layers_test.keys():
-            output_layers_test[layer].to_csv(os.path.join(args.outdir, '.'.join([args.prefix, 'test_decoded', layer, 'csv'])), header=True)
-
-
-    # evaluate off-the-shelf methods on the main target variable
-    if args.evaluate_baseline_performance:
```
Comment (borauyar) [inline flexynesis/__main__.py:550]: please don't remove baseline evaluation chunk

**Comment by amitpande74** [comment]
✅ Update summary for feat/inference-mode

I tested the branch end-to-end on Colab Pro and confirmed that the main issues are fixed:

CLI documentation restored
The main() docstring is back.
python -m flexynesis --help shows the full set of training and inference flags again.
Baseline evaluation preserved
RandomForest, SVM, and XGBoost baselines now run inline when --evaluate_baseline_performance is set.

Outputs:
baseline.stats.csv
baseline.predicted_labels.csv
Inference early-exit path works
When --pretrained_model + --artifacts + --data_path_test are provided, training is skipped.
Inference stub runs cleanly and prints the correct artifact + data paths.
Artifacts writing after training
After deep model training, an *.artifacts.joblib file is written alongside the other outputs.
This makes inference reproducible and keeps the outputs consistent.

🔬 Verification runs

Baseline-only
[INFO] Running baseline evaluation on target 'Erlotinib' (RandomForest/SVM/XGBoost)
Training method: RandomForest
Training method: SVM
Training method: XGBoost
[INFO] Baseline evaluation finished and files saved.


Deep + baseline
Embeddings, feature logs, final model, baseline stats/preds, and artifacts were all saved.

Example:
[INFO] Wrote inference artifacts to /content/out_train_with_baseline/deep_plus_baseline.artifacts.joblib
[INFO] Finished the analysis!


Inference-only
[flexynesis] Inference mode stub reached ✅
  artifacts: /content/out_train/smoke.artifacts.joblib
  data_path_test: /content/data/dataset1
  outdir/prefix: /content/out_infer / infer_smoke

📂 Final outputs confirmed

For a deep+baseline run, the following files were produced in outdir:

*.artifacts.joblib
*.baseline.predicted_labels.csv
*.baseline.stats.csv
*.embeddings_train.csv
*.embeddings_test.csv
*.feature_logs.*.csv
*.feature_importance.IntegratedGradients.csv
*.final_model.pth
*.predicted_labels.csv
*.stats.csv

**Comment by borauyar** [comment]
Thank you @amitpande74 !
